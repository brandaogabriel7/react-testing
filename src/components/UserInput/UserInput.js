import React from 'react';

const UserInput = ({changed, initialValue}) => {
    return <input type="text" onChange={changed} defaultValue={initialValue}/>;
};

export default UserInput;