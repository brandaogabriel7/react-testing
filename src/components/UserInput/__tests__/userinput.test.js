import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import UserInput from '../UserInput';

const TEXTBOX_ROLE = 'textbox';

const initialValueTestCases = [
    ['batata'],
    ['random initial value'],
    ['another random initial value']
];

describe('user input should', () => {
    it('render correctly', () => {
        render(<UserInput />)
        expect(screen.getByRole(TEXTBOX_ROLE)).toBeInTheDocument();
    });

    it('match snapshot', () => {
        render(<UserInput />)
        expect(screen.getByRole(TEXTBOX_ROLE)).toMatchSnapshot();
    });

    it('call the changed handler when input changes', () => {
        const mockChangedHandler = jest.fn();
        render(<UserInput changed={mockChangedHandler} />);
        
        const typedText = 'opaaa';
        userEvent.type(screen.getByRole(TEXTBOX_ROLE), typedText);
        expect(mockChangedHandler).toHaveBeenCalledTimes(typedText.length);
    });

    it.each(initialValueTestCases)('have initial value of \'%s\'', (initialValue) => {
        render(<UserInput initialValue={initialValue} />);

        const userInput = screen.getByRole(TEXTBOX_ROLE);
        expect(userInput).toBeInTheDocument();
        expect(userInput).toHaveValue(initialValue);
    });
});