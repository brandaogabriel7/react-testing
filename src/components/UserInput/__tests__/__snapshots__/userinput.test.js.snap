// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user input should match snapshot 1`] = `
<input
  type="text"
  value=""
/>
`;
